azkaban-common/src/main/java/azkaban/project/AzkabanProjectLoader.java

```diff
@@ -79,10 +79,53 @@
       final File archive, final String fileType, final User uploader, final Props additionalProps)
       throws ProjectManagerException {
     log.info("Uploading files to " + project.getName());
+    final Map<String, ValidationReport> reports;
+
+    // Since props is an instance variable of ProjectManager, and each
+    // invocation to the uploadProject manager needs to pass a different
+    // value for the PROJECT_ARCHIVE_FILE_PATH key, it is necessary to
+    // create a new instance of Props to make sure these different values
+    // are isolated from each other.
+    final Props prop = new Props(this.props);
+    prop.putAll(additionalProps);
 
-    // Unzip.
     File file = null;
     try {
+      // Unzip the project.
```

Comment:
Consider breaking up this method to smaller units.

Reply:
Will remove, thanks!